llvm/lib/Target/AIE/AIE2GenInstrInfo.td

```diff
@@ -201,12 +202,16 @@ let mayLoad = true, mayStore = true in {
 }
 let mayLoad = true, mayStore = false in {
   let Itinerary = II_LDA in {
-  def LDA_dms_lda_idx : AIE2_dms_lda_inst_lda< (outs OP_mLdaScl:$mLdaScl) , (ins eP:$ptr, eDJ:$dj),
-      "lda", "$mLdaScl, [$ptr, $dj]">, 
-    AIE2_ag_all__ag_idx;
-  def LDA_dms_lda_idx_imm : AIE2_dms_lda_inst_lda< (outs OP_mLdaScl:$mLdaScl) , (ins eP:$ptr, imm6x4:$imm),
-      "lda", "$mLdaScl, [$ptr, $imm]">, 
-    AIE2_ag_all__ag_idx_imm;
+  let ItineraryRegPairs = [ItinRegClassPair<II_LDA_R,[OperandRegClass<0, eR>]>, ItinRegClassPair<II_LDA_P,[OperandRegClass<0, eP>]>, 
+                           ItinRegClassPair<II_LDA_M,[OperandRegClass<0, eM>]>, ItinRegClassPair<II_LDA_DC,[OperandRegClass<0, eDC>]>,
+                           ItinRegClassPair<II_LDA_DJ,[OperandRegClass<0, eDJ>]>, ItinRegClassPair<II_LDA_DN,[OperandRegClass<0, eDN>]>] in {
```

Comment:
I think LDA can also write to `lr`. I guess in that case we'll be conservative and book all ports?

Reply:
Yes, anytime any of `ItineraryRegPairs` is not matched we fall back to the conservative and reserved according to default Itinerary, in this case `II_LDA`